docs/_layouts/global.html

```diff
@@ -207,14 +205,79 @@ <h1 class="title">{{ page.title }}</h1>
     })
 </script>
 
+<!-- Pure JavaScript implementation of ScrollSpy -->
 <script>
-    var controller = new ScrollMagic.Controller();
+    //finds toc and automatically sorts by main list items as parents and nested list items as children
 
-    new ScrollMagic.Scene({
-        offset: 20
-    })
-    .setClassToggle(".navbar-inner", "shrunk")
-    .addTo(controller);
+    var toc = document.getElementById("toc");
+    if(toc != null) {
+        var topPadding = 12; // amount of space above header to start noticing a section
+        var navBar = document.getElementById("topbar").offsetHeight;
+        var tocEl = toc.nextElementSibling.getElementsByTagName("li");
+        var sections = toc.nextElementSibling.getElementsByTagName("a");
+        sections = Array.prototype.slice.call(sections);
+        var sectionsLen = sections.length;
+        var parentIndex;
+        var parentTag = sections[0].getAttribute("href") == "#" ? "H1" : document.getElementById(sections[0].getAttribute("href").substring(1)).nodeName;
+        for(var i = 0; i < sectionsLen; i++) {
+            //[element, offset from top, group]
+            var element = sections[i].getAttribute("href") == "#" ? document.getElementById("content").firstElementChild : document.getElementById(sections[i].getAttribute("href").substring(1));
+            if(element.nodeName == parentTag) {
+                parentIndex = i;
+            }
+            sections[i] = [element, element.offsetTop - navBar - topPadding, parentIndex];
+        }
+        var active = 0;
+        tocEl[0].classList.toggle("active");
+        var sublist;
+        var sublistPos = 0;
+        var sublistChildren = 0;
+        var locked = false;
+        var runOnScroll =  function(e) {
+            if(locked) return;
+            locked = true;
+            var pos = 0;
+            //get current active element
+            for(var i = 0; i < sectionsLen; i++) {
+                if(window.scrollY > sections[i][1]) {
+                    pos = i;
+                }
+                else {
+                    break;
+                }
+            }
+
+            if(active != pos) { // change in active element
+                tocEl[active].classList.remove('active');
+                active = pos;
+                tocEl[active].classList.add('active');
+                if(sublist != null) {
```

Comment:
Are only 1 level of sublists supported? What happens if a toc has more than 2 levels?